lib/request.js

```diff
@@ -109,7 +110,9 @@ function Request (options) {
     this.headers[field.toLowerCase()] = '' + value
   }
 
-  this.headers['user-agent'] = this.headers['user-agent'] || 'lightMyRequest'
+  if (!this.headers['user-agent'] && autoFillHeaders) {
```

Comment:
I copy pasted it. Removing the autoFillHeaders

Reply:
I can't assign an `undefined` to a header - it will appear in this form in `req.headers` and an error will occur

Reply:
Will this code work?

```js
if ('user-agent' in this.headers === false && autoFillHeaders) {
  this.headers['user-agent'] = 'lightMyRequest'
}
```

Reply:
have refactored to this form

Reply:
this form of verification

```js
'user-agent' in this.headers === false 
```
allows the presence of a header with the value undefined - this will not let you know that the header is not filled - so refer to original code with check 

```js
!req.headers['user-agent']
```

Reply:
Until I understand what to do with the situation when the header is explicitly defined as `undefined` in case when `autoFillHeaders = false` - this causes an error:

```
✖ invalid value "undefined" for header user-agent

    const value = headers[field]
    assert(value !== undefined, 'invalid value "undefined" for header ' + field)
----^
    this.headers[field.toLowerCase()] = '' + value
  }
```

Reply:
My Commit https://github.com/fastify/light-my-request/compare/master...Uzlopak:light-my-request:make-special-headers-undefinable is passig all the unit tests and does exactly what you want to achieve.

I just dont want that you think that I am a showoff after you invested your time to come with the first solution, and now i come around the corner and say: "Hey look this is doing the same but supposedly better".

Reply:
curl does not set implicity `'user-agent': 'undefined'` when someone missed to add this header to the request:

```
curl -X POST http://localhost:3000/route --body "{}"
```

Reply:
This is a valid argument. But if you consider light-my-request more like a browser, than browsers have default user-agent headers, which can be overwritten.